records/state/state.go

```diff
@@ -162,7 +162,8 @@ func statusIPs(st []Status, src func(*Status) []string) []string {
 	// https://github.com/apache/mesos/blob/0.24.0/src/slave/slave.cpp#L5226-L5238
 	ts, j := -1.0, -1
 	for i := range st {
-		if st[i].State == "TASK_RUNNING" && st[i].Timestamp > ts {
+		state := st[i].State
+		if (state == "TASK_RUNNING" || state == "TASK_STAGING") && st[i].Timestamp > ts {
```

Comment:
Yeah, although Mesos-DNS can't guarantee the health of a record. How long do you think a task can be in staging state? What would be the implications of clients connecting to services in that state? Wouldn't they need to retry in case of failure anyway?